app/src/pages/authoring/index.tsx

```diff
@@ -16,39 +27,74 @@ const useStyles = createStyles(() => ({
 
 export default function AuthoringPage() {
   const [resourceType, setResourceType] = useState<ArtifactResourceType>('Measure');
+  const [selectedArtifact, setSelectedArtifact] = useState<string | null>(null);
+
+  const {
+    data: artifacts,
+    isLoading: artifactsLoading,
+    error: artifactError
+  } = trpc.service.getArtifactsByType.useQuery({ resourceType });
 
   const ctx = trpc.useContext();
   const { classes } = useStyles();
+  const router = useRouter();
+
+  const successNotification = (data: { draftId: string }, createdFromArtifact: boolean) => {
+    const message = createdFromArtifact
+      ? `Draft of ${resourceType}/${selectedArtifact} successfully created`
+      : `${resourceType} successfully created`;
+    notifications.show({
+      title: `${resourceType} Created!`,
+      message: message,
+      icon: <CircleCheck />,
+      color: 'green'
+    });
+    router.push(`authoring/${resourceType}/${data.draftId}`);
+    ctx.draft.getDraftCounts.invalidate();
+  };
+
+  const errorNotification = (errorMessage: string, createdFromArtifact: boolean) => {
+    const message = createdFromArtifact
+      ? `Attempt to create draft of ${resourceType}/${selectedArtifact} failed with message: ${errorMessage}`
+      : `Attempt to create ${resourceType} failed with message: ${errorMessage}`;
+    notifications.show({
+      title: `${resourceType} Creation Failed!`,
+      message: message,
+      icon: <AlertCircle />,
+      color: 'red'
+    });
+  };
 
   const draftMutation = trpc.draft.createDraft.useMutation({
     onSuccess: data => {
-      notifications.show({
-        title: `${resourceType} Created!`,
-        message: `${resourceType} successfully created`,
-        icon: <CircleCheck />,
-        color: 'green'
-      });
-      router.push(`authoring/${resourceType}/${data.draftId}`);
-      ctx.draft.getDraftCounts.invalidate();
+      successNotification(data, false);
     },
     onError: e => {
-      notifications.show({
-        title: `${resourceType} Creation Failed!`,
-        message: `Attempt to create ${resourceType} failed with message: ${e.message}`,
-        icon: <AlertCircle />,
-        color: 'red'
-      });
+      errorNotification(e.message, false);
     }
   });
 
-  const router = useRouter();
+  const draftFromArtifactMutation = trpc.service.convertArtifactById.useMutation({
+    onSuccess: data => {
+      successNotification(data, true);
+    },
+    onError: e => {
+      errorNotification(e.message, true);
+    }
+  });
 
   const createResource = () => {
     const newResource = resourceType === 'Measure' ? { ...MeasureSkeleton } : { ...LibrarySkeleton };
     newResource.id = uuidv4();
     draftMutation.mutate({ resourceType, draft: newResource });
   };
 
+  const createDraftArtifactOf = () => {
```

Comment:
This is a nitpick but since this `const` does not take in any arguments, I feel like it might be more appropriate to name this `createDraftArtifact` rather than saying "of" at the end. Or we could make it more descriptive and call it `createDraftArtifactOfResource`